gobblin-runtime/src/main/java/gobblin/runtime/MysqlDatasetStateStore.java

```diff
@@ -64,12 +94,57 @@ public MysqlDatasetStateStore(DataSource dataSource, String stateStoreTableName,
    * @throws IOException if there's something wrong reading the {@link JobState.DatasetState}s
    */
   public Map<String, JobState.DatasetState> getLatestDatasetStatesByUrns(String jobName) throws IOException {
-    List<JobState.DatasetState> previousDatasetStates =
-        getAll(jobName, "%" + CURRENT_DATASET_STATE_FILE_SUFFIX + DATASET_STATE_STORE_TABLE_SUFFIX, true);
+    final String storeName = jobName;
```

Comment:
I can reduce the duplication between MysqlStateStore and MysqlDatasetStateStore pretty easily.  Can you explain further your problem with exposing the querying logic to MysqlDatasetStateStore?  I don't see any easy way to get the latest dataset states broken out by Urn, unless we allow the MysqlDatasetStateStore  to do the query or push that functionality into MysqlStateStore and allow MysqlDatasetStateStore to delegate the call.